parse_str_test.go

```diff
@@ -4,27 +4,45 @@ import (
 	"fmt"
 	"reflect"
 	"testing"
+
+	"github.com/elliotchance/orderedmap/v2"
 )
 
+type dict map[any]any
+
+func convertOrderedMapToMap(omap orderedmap.OrderedMap[any, any]) dict {
+	// Convert OrderedMap to map
+	comparison := make(dict, omap.Len())
+	for el := omap.Front(); el != nil; el = el.Next() {
+		switch v := el.Value.(type) {
+		case orderedmap.OrderedMap[any, any]:
+			comparison[el.Key] = convertOrderedMapToMap(v)
+		default:
+			comparison[el.Key] = el.Value
+		}
+	}
+	return comparison
+}
```

Comment:
As [@smc0210's comment](https://github.com/portone-io/gophplib/pull/16#discussion_r1537217909), depending on the version of go you're running, `fmt.Println` a map produces different results. It's assumed that the actual use of `fmt.Println` for `ParseStr` function's outputs would mainly be in logging or test code contexts. Therefore, while keeping the `convertOrderedMapToMap` for Example functions, I've added [a NOTE in comments](https://github.com/portone-io/gophplib/pull/16/commits/eb30112a6c685d84241c485a3255ba3c299a0b17) for cases where `fmt.Println` results vary by version to use these comments for documentation purposes if similar issues occur in the future.

Reply:
https://github.com/portone-io/gophplib/pull/16#discussion_r1535153663

What I initially meant to say was that you need to change the map uses internally.
Now because `d` of `phpSymtable` is replaced by `orderedmap.OrderedMap[any, any]`, the order will be preserved.

By the way, if the `fmt.Println` function prints map in arbitrary order, is it okay to keep the validation way of existing test cases like below...?

```go
fmt.Println(convertOrderedMapToMap(ParseStr("dict[k1][k2]=v1&dict[k1][k3]=v2")))
// map[dict:map[k1:map[k2:v1 k3:v2]]]
```

What I'm wondering is, if the printed result of `fmt.Println` varies between Golang versions, shouldn't you change the way you validate test cases.

Reply:
If example function has "Output:" comment, [it tries to run and verify example code](https://pkg.go.dev/testing#hdr-Examples).

> The package also runs and verifies example code. Example functions may include a concluding line comment that begins with "Output:" and is compared with the standard output of the function when the tests are run.

So if you want to verify the result of example, you need to ensure that the example output is the same.
If you do not want to verify the result, then you need to delete the "Output:" comment part.

Reply:
How about removing "Outputs:" comment and adding the expected output value right below the case? 
Like this
```go
func ExampleParseStr(){
    // Plain key-value
    fmt.Println(convertOrderedMapToMap(ParseStr("key=value&foo=bar")))
    // map[foo:bar key:value]

    // Array will be parsed as map with integer keys
    fmt.Println(convertOrderedMapToMap(ParseStr("arr[0]=A&arr[1]=B&arr[2]=C")))
    // map[arr:map[0:A 1:B 2:C]]
    ...
}

```

Reply:
I think that would be good too.

Reply:
I've added the Example function cases, which were not in the existing test cases in the `TestParseStr` function.  Also removed the `Outputs:` annotation from the Example function and added the expected value to the comment for each case.